pkg/server/api.go

```diff
@@ -186,31 +187,33 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 type healthHandler struct{}
 
-// deriveSpecVersionFromUseragent returns a supported Ignition config spec version for a given Ignition user agent
-// or an error for unsuported user agents.
-// Only Ignition versions that come with support for either spec v2.2 or spec v3.1 are supported
-func deriveSpecVersionFromUseragent(useragent string) (*semver.Version, error) {
-	// for now, serve v2 if we receive a request without a useragent header with Ignition
-	// this happens if the user pings the endpoint directly (e.g. with curl)
-	// and we don't want to break existing behaviour
-	if !strings.HasPrefix(useragent, "Ignition/") {
+// deriveSpecVersionFromAcceptHeaderUseragent returns a supported Ignition config spec version for a given Accept header.
+// For non-Ignition Accept headers it defaults to config spec v2.2.0
+func deriveSpecVersionFromAcceptHeader(acceptHeader string) (*semver.Version, error) {
+	// serve v3 if we receive a request without an Ignition accept header.
```

Comment:
whoops, shouldn't have changed - we're still serving spec 2 by default.
@runcom @yuqi-zhang now that everything in 4.6 has made the switch to v3, shouldn't we serve it here by default now, too? (might be a bit weird ergonomically if you curl for the config, expecting you get what your new nodes will get, but instead you'll receive the down-translated conf.)

Reply:
Hmm the "serve by default" is there to maintain backwards compatibility. If someone is not using ignition directly, we're giving them time to make the transition from spec 2 to spec 3 by not breaking existing behaviour. In that sense I think its better to serve spec2 here so as not to break existing CI, etc.

Reply:
But don't we want to encourage folks to migrate their custom Ignition to v3 too? If we keep serving v2 we may not encourage that. (If you'd really like it to stay at v2 however, then I'm OK with it :)

Reply:
This'll be discussed in a Cabal. Let's leave current behaviour (v2) for now

Reply:
I wasn't aware this was agreed with the Windows team. Sticking v2.2 default for now :)